### PR TITLE
react query implemented

### PR DESCRIPTION
**Pull request by gatinhap**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -7,21 +7,26 @@ import { theme } from "./theme/theme.ts";
 import { ThemeProvider } from "styled-components";
 import "react-toastify/dist/ReactToastify.min.css";
 import { StyledToastContainer } from "./components/toast/Toast.styles.ts";
+import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 
 const App = () => {
+  const queryClient = new QueryClient();
+
   return (
     <>
```
Comment (michal-karbasz) [inline src/App.tsx:16]: seems redundant

File: src/components/form/Form.tsx [hunk 2]
```diff
@@ -27,9 +28,13 @@ const Form = () => {
       });
   };
 
+  const { mutate } = useMutation({
```
Comment (michal-karbasz) [inline src/components/form/Form.tsx:31]: FYI, destructuring may be ok if you have only one mutation, but in larger applications, it's usually a case that you have multiple. It's better to name the mutation and then use it like `mutationName.mutate`

File: src/components/form/Form.tsx [hunk 3]
```diff
@@ -27,9 +28,13 @@ const Form = () => {
...
+    mutationFn: (newPlant: FormValues) => onSubmit(newPlant),
+  });
+
   return (
     <>
```
Comment (michal-karbasz) [inline src/components/form/Form.tsx:36]: redundant fragment

File: src/components/plantCollection/PlantCollection.tsx [hunk 4]
```diff
@@ -1,32 +1,42 @@
...
+  }
+
+  if (isError) {
+    return <Text variant={"large"}>Error: {error.message}</Text>;
+  }
+
+  if (data) {
+    return (
+      <PlantCollectionStyled>
+        {data.map((plant) => {
```
Comment (michal-karbasz) [inline src/components/plantCollection/PlantCollection.tsx:30]: redundant brackets

File: src/pages/plantDetailPage/PlantDetailPage.tsx [hunk 5]
```diff
@@ -3,103 +3,120 @@ import NavItem from "../../components/navItem/NavItem.tsx";
...
-    JSON.parse(localStorage.getItem("plantsList") || "") || [];
-
   const { plantId } = useParams();
 
-  const currentPlantId = plantsCollection.find(
-    (singlePlant) => singlePlant.id === plantId,
-  );
+  const getPlantsList = async () => {
+    const res = await pb.collection(PLANTS_COLLECTION).getList(1, 100);
+    return res.items;
```
Comment (michal-karbasz) [inline src/pages/plantDetailPage/PlantDetailPage.tsx:18]: new line

File: src/pages/plantDetailPage/PlantDetailPage.tsx [hunk 6]
```diff
@@ -3,103 +3,120 @@ import NavItem from "../../components/navItem/NavItem.tsx";
...
 const PlantDetailPage = () => {
-  const plantsCollection: FormValues[] =
-    JSON.parse(localStorage.getItem("plantsList") || "") || [];
-
   const { plantId } = useParams();
 
-  const currentPlantId = plantsCollection.find(
-    (singlePlant) => singlePlant.id === plantId,
-  );
+  const getPlantsList = async () => {
```
Comment (michal-karbasz) [inline src/pages/plantDetailPage/PlantDetailPage.tsx:16]: it's duplicated, but more importantly you shouldn't need to fetch the entire list here. Is there another method in pb that let's you extract only one item from a collection? That's the only thing you need here?

This is important for caching reasons and performance in the long run. 
If you fetch only one element via react-query and save it as a single entry in cache then when you modify/delete (mutate in general) that item you don't need to refetch the entire list of plants again. You will only tell the cache that this one plant has e.g. different watering requirements.

File: src/components/form/Form.tsx (hunk 2 above)
Comment (gatinhap) [reply src/components/form/Form.tsx:31]: oh good to know!

File: src/components/plantCollection/PlantCollection.tsx (hunk 4 above)
Comment (gatinhap) [reply src/components/plantCollection/PlantCollection.tsx:30]: you mean around `(plant)`? I've tried to remove them but my code editor brings them back😅

File: src/pages/plantDetailPage/PlantDetailPage.tsx [hunk 7]
```diff
@@ -3,103 +3,120 @@ import NavItem from "../../components/navItem/NavItem.tsx";
...
+  if (isPending) {
+    return <Text variant={"large"}>Loading...</Text>;
+  }
+
+  if (isError) {
+    return <Text variant={"large"}>Error: {error.message}</Text>;
+  }
+
+  const currentPlant =
+    data && data.find((singlePlant) => singlePlant.id === plantId);
```
[the remark this answers is not in the document]
Comment (gatinhap) [reply src/pages/plantDetailPage/PlantDetailPage.tsx:35]: removed

File: src/components/plantCollection/PlantCollection.tsx (hunk 4 above)
Comment (michal-karbasz) [reply src/components/plantCollection/PlantCollection.tsx:30]: after fat arrow :) you don't need an explicit `return` here

**Review by michal-karbasz — APPROVED** [review]

**Pull request merged**